web3.js/src/transaction.ts

```diff
@@ -172,8 +191,22 @@ export class Transaction {
   /**
    * Construct an empty Transaction
    */
-  constructor(opts?: TransactionCtorFields) {
-    opts && Object.assign(this, opts);
+  constructor(opts?: TransactionCtorFields | TransactionCtorFields_DEPRECATED) {
+    if (!opts) {
+      return;
+    } else if (Object.prototype.hasOwnProperty.call(opts, 'recentBlockhash')) {
+      console.warn(
+        'Creating a Transaction with `recentBlockhash` is deprecated. ',
+      );
```

Comment:
This is good, but this warning will be delivered at runtime. Let's instead deliver the warning at typecheck time! Check out the TypeScript playground link I dropped [here](https://github.com/solana-labs/solana/issues/23949#issuecomment-1090612996) for an example of how to make TypeScript literally cross out deprecated invocations of this constructor in your editor, with a helpful message about what to do instead.

Reply:
Ah, I see. Thanks! I'll take another look and make sure to get this updated.

Reply:
@steveluscher following your method makes the `Transaction` type unrecognizable outside of `transaction.ts`

<img width="1127" alt="image" src="https://user-images.githubusercontent.com/46392676/164345385-18464d7e-83c4-4f5e-8087-b8b33ed97372.png">

I'm a little uncertain of the best practice to approach this since TS does not allow multiple constructors.

Reply:
Interesting! Do you have this checked in somewhere that I can take a look at?

Reply:
@steveluscher for sure, I pushed it [here](https://github.com/MLH-Fellowship/solana/tree/deprecate-transaction-ctor) !

Reply:
Solved it! My initial take was way too complicated. ([link](https://www.typescriptlang.org/play?#code/C4TwDgpgBAQgNgewMYGsAWBDAzmqBeKLYAJwEsA7AcwG4AoW0SKAFWI3KwyWFIXIGFgCYgDFSEOABMsAfQAiAUQAKAJQX8AgswVz8UAPIBbUsAA8rdp268BQ0eKlYANFADkcDMAhF4ydNjRXAD4oADIoAG9aKChiCCQIcmBfVEwcAH4ALlhEVICoAB8ocgBXODg6AF86BnBoCw4uHj5BYTEJaT0omI8vH1z-DOzumNGAIwG0tGyUwbRqKGjRnuxgADUMOFJJWYAJCFJKNGBs0sMxiGIl6qWEYDRLgGVgEoAzV6zCEgoaWhvaJAeLBYKD8PjkeLNciRJYyIHrTbbPYHI4nYolc6XQpQErkSQQV4UCCSOgxGRxBJJWZTGaTfJFUrlbG4-GEiEk+gxAD0XKgcgJRKg92gFC8xFeXGgr2EQoeUFeuOsfCFdSgAHcTLgMFAtkQoAhXvqAG6XRAYaRLJZcgBU1qgAAF8WAKZ5iUsoHaAAYUxLJOk4T1QNJQC6JKBOl1eSQAOnddqUcAg2GgYGwIM9vW8fr8NMiEz8uwCtILARc8I2Wx2A32h2OpwxF2IlUDFCISZjcZYaFIII1TIu4YgzvirskUAoUG1AA1owAGOdQE3ELA2fWG+1YBAecgYLlqiBjADM0YAVlhY6NrVylkg+ERiCVuMIABQIMDALCfFhsRpK2xtBxpHkZQ1E0bQ5AASlJRZuV5Zg5QhNVCEOHcXjiG87xIR87Ffd9P2yBorChVp7A6LAoM5KAeS7MNWUFYVx0MMBE0MX1PFXFdKFQkp0JiW8OCwp9iFwj9PkIpobBI9pHGxcS-ykwDZEUVR1C0HQIJhZYYlIQ1nwAQjfD8NJGLTRjiNDyGg5ZKndUYdKgETzzSfQ1XIJRiDfS5QGfVwfSpf1Agg4zbOWaiAElgFcEEGK3SQAFoiBARN9XfGxzxC0Z+M3RNozVDBiHIHz+DidiqEnKBPTBcgIT-QMNXuCq-OzPIA3HEEIxHKNo261wKNMvjMP1KR9DwvRDJBbBv0sCSWjsaSgOU0C1LkKytPuHto3JeJfWpfICFikaP2jJrdpwVaYkqKAJCwaATK08LIuihCICQxLksMtKL36rLgGKF7DpBfbRsmuTiLmxTztGdbzzhVYKyRasUWOPREIB6NM36HMAnSdG4cRKsCyR4BIZiaHNpOgKUf+vDcb6Zq5hx-MWvmDKbNM64-noH6hskKqaqhKmkL5yEbGfCIKaxnBslcDAxiQVxKgo7nEOFv9BdBcERb4MWMfp3MIiZuZpdl+WyzxytkVrNEAEZZztypFeoIA))